dts/arm/ambiq/ambiq_apollo3_blue.dtsi

```diff
@@ -0,0 +1,171 @@
+/* SPDX-License-Identifier: Apache-2.0 */
+
+#include <arm/armv7-m.dtsi>
+#include <mem.h>
+#include <freq.h>
+#include <zephyr/dt-bindings/i2c/i2c.h>
+#include <zephyr/dt-bindings/gpio/gpio.h>
+
+/ {
+	clocks {
+		uartclk: apb-pclk {
+			compatible = "fixed-clock";
+			clock-frequency = <DT_FREQ_M(24)>;
+			#clock-cells = <0>;
+		};
+	};
+
+	cpus {
+		#address-cells = <1>;
+		#size-cells = <0>;
+
+		cpu0: cpu@0 {
+			compatible = "arm,cortex-m4f";
+			reg = <0>;
+		};
+	};
+
+	/* Flash region */
+	flash0: flash@C000 {
+		compatible = "soc-nv-flash";
+		reg = <0x0000C000 0xF4000>;
+	};
+
+	/* TCM */
+	tcm: tcm@10000000 {
+		compatible = "zephyr,memory-region";
+		reg = <0x10000000 0x10000>;
+		zephyr,memory-region = "ITCM";
+	};
+
+	/* SRAM */
+	sram0: memory@10010000 {
+		compatible = "mmio-sram";
+		reg = <0x10010000 0x50000>;
+	};
+
+	soc {
+		compatible = "ambiq,apollo3-blue", "ambiq,apollo3x", "simple-bus";
+
+		pwrcfg: pwrcfg@40021000 {
+			compatible = "ambiq,pwrctrl";
+			reg = <0x40021000 0x400>;
+			#pwrcfg-cells = <2>;
+		};
+
+		stimer0: stimer@40008140 {
+			compatible = "ambiq,stimer";
+			reg = <0x40008140 0x80>;
```

Comment:
Got a Doubt while Looking at the Data Sheet of the Apollo3 Blue SOC , In some Nodes, the size mentioned in the reg property are not matches with the Data Sheet. For example, in stimer the dataa sheet says that the stimer registers ends at 0x4000830C but here it is mentioned as `reg = <0x40008140 0x80>;` which says it ends at 0x400081B9.
Is there any reason for this mismatch or is it by mistake?

Reply:
@srisurya1 thank you for the comment, the reg size parameter here was a copy error that derived from apollo4plus soc, however it's not been used in our driver, moreover we will use macros defined in HAL to replace all these addresses and sizes, change will soon be merged.

Reply:
@AlessandroLuo Kindly, Check other nodes also for the size, before your next merge. Thanks For the Quick response :)